pkg/volume/gce_pd/gce_util.go

```diff
@@ -39,8 +41,17 @@ func (util *GCEDiskUtil) AttachAndMountDisk(pd *gcePersistentDisk, globalPDPath
 	if err != nil {
 		return err
 	}
-	if err := gce.(*gce_cloud.GCECloud).AttachDisk(pd.pdName, pd.readOnly); err != nil {
-		return err
+	i := 0
+	t := time.Second
+	for i < attachRetries {
+		i++
+		if err := gce.(*gce_cloud.GCECloud).AttachDisk(pd.pdName, pd.readOnly); err == nil {
+			break
+		} else if i == attachRetries {
+			return err
+		}
+		time.Sleep(t)
+		t = t * t
```

Comment:
you init t to 1 second, but it is stored as nanoseconds.  The second time through you will sleep for 1B * 1B nanoseconds, or approximately 32 years.  Third time through, will probably outlast the universe.

Maybe you meant t = t*2 ?  But even that, is it right to sleep and retry here, rather than just letteing the resync loop catch it?  I'm not sure from context whether this runs in a per-pod goroutine, or whether it will block all pods syncs for up to a minute.